### ISSUE TITLE
GDAL build is failing on MacOS with Julia 1.6.0-beta1

### ISSUE THREAD

**Issue by vlandau** [JuliaGeo/GDAL.jl#113]
It looks like it is maybe an upstream issue with MbedTLS, which @visr pointed out on Slack. Here's the full error:
```julia
ERROR: LoadError: InitError: could not load library "/Users/runner/.julia/artifacts/2c3cdffc520e54030c005f3ea5b7b465a5a5205a/lib/libgdal.28.dylib"
dlopen(/Users/runner/.julia/artifacts/2c3cdffc520e54030c005f3ea5b7b465a5a5205a/lib/libgdal.28.dylib, 1): Library not loaded: @rpath/libmbedtls.12.dylib
  Referenced from: /Users/runner/.julia/artifacts/2c3cdffc520e54030c005f3ea5b7b465a5a5205a/lib/libgdal.28.dylib
  Reason: image not found
```

The full stacktrace/run environment info can be found [here](https://github.com/Circuitscape/Omniscape.jl/runs/1682753261?check_suite_focus=true#step:6:121).

**Comment by visr** [comment]
Indeed GDAL.jl CI also picked this up: https://github.com/JuliaGeo/GDAL.jl/runs/1640642495?check_suite_focus=true

Cause and solution seems to be: https://github.com/JuliaLang/julia/issues/39015#issuecomment-751690888. Though it is unfortunate that will require a different build for different julia version.

**Comment by zhangliye** [comment]
For the Julia issule [JuliaLang/julia#39015 (comment)](https://github.com/JuliaLang/julia/issues/39015#issuecomment-751690888), it is said that it requires package update as follows.

> Thanks, so not really a release blocker then since it just requires package updates. [JuliaLang/julia#39015 (comment)](https://github.com/JuliaLang/julia/issues/39015#issuecomment-751693052)

**Comment by visr** [comment]
It's not so simple, though, especially if we don't want to only support julia 1.6+ going forward. NetCDF has the same issue, see also the discussion in https://github.com/JuliaPackaging/Yggdrasil/pull/2402.

**Comment by visr** [comment]
We should follow the approach used to resolve this same issue by NetCDF, so basically apply the setup used in https://github.com/JuliaPackaging/Yggdrasil/tree/master/N/NetCDF, see also https://github.com/JuliaPackaging/Yggdrasil/issues/2401.

First this should be done for PROJ_jll, then we can do it for GDAL_jll. I'm scarce on time at the moment (:baby:), so help would be appreciated such that this can be fixed before 1.6 final is released.
